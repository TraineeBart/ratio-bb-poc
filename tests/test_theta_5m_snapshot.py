# ╭──────────────────────────────────────────────────────────────╮
# │ File: tests/test_theta_5m_snapshot.py                        │
# │ Module: Snapshot Testing                                     │
# │ Doel: Snapshot-validatie van verrijkte theta 5m dataset      │
# │ Auteur: Quality EngineerGPT                                  │
# │ Laatste wijziging: 2025-07-01                                │
# │ Status: stable                                               │
# ╰──────────────────────────────────────────────────────────────╯

import pandas as pd
from pathlib import Path

def test_theta_5m_column_headers_snapshot(snapshot):
    """
    🧠 Functie: test_theta_5m_column_headers_snapshot
    Verifieert of de kolomstructuur van de verrijkte theta 5m CSV overeenkomt met de opgeslagen snapshot.

    ▶️ In:
        - snapshot: Syrupy snapshot fixture

    ⏺ Out:
        - Assertion tegen bekende kolomlijst
    """
    file_path = Path("/opt/ratio-bb-poc/data/signals_bb_theta_5m.csv")
    df = pd.read_csv(file_path)
    actual_columns = list(df.columns)

    assert actual_columns == snapshot