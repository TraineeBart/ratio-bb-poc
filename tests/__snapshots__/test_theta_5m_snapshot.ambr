# serializer version: 1
# name: test_theta_5m_column_headers_snapshot
  list([
    'timestamp',
    'open',
    'high',
    'low',
    'close',
    'volume',
    'rsi',
    'sma_rsi',
    'sma_9',
    'sma',
    'upper_band',
    'lower_band',
    'ratio_lower',
    'ratio_upper',
    'signal',
  ])
# ---
